src/internal/SetupResumeAcceptor.cpp

```diff
@@ -143,8 +143,19 @@ void SetupResumeAcceptor::processFrame(
         break;
       }
 
-      removeConnection(transport);
-      onResume(std::move(transport), std::move(resumeParams));
+      if (!onResume(transport, std::move(resumeParams))) {
+        transport->outputFrameOrEnqueue(frameSerializer->serializeOut(
+            Frame_ERROR::connectionError("Resumption Failure")));
```

Comment:
(y) Also "Resumption Failure" seems a bit dishonest, can it be "Resumption not supported" instead?